src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  HeaderSectionItem,
+  LinkSectionItem,
+  Section,
+  ToggleSectionItem,
+} from '@atb/components/sections';
+import {StyleSheet, Theme} from '@atb/theme';
+import {ProfileTexts, useTranslation} from '@atb/translations';
+import React from 'react';
+import {ScrollView} from 'react-native-gesture-handler';
+import {FullScreenHeader} from '@atb/components/screen-header';
+import {Linking, View} from 'react-native';
+import PrivacySettingsTexts from '@atb/translations/screens/subscreens/PrivacySettingsTexts';
+import {useBeacons} from '@atb/beacons/use-beacons';
+import {Button} from '@atb/components/button';
+import {Delete} from '@atb/assets/svg/mono-icons/actions';
+import {destructiveAlert} from './utils';
+import {useRemoteConfig} from '@atb/RemoteConfigContext';
+import {useSearchHistory} from '@atb/search-history';
+
+export const Profile_PrivacyScreen = () => {
+  const {t} = useTranslation();
+  const {
+    kettleInfo,
+    onboardForBeacons,
+    revokeBeacons,
+    deleteCollectedData,
+    isBeaconsSupported,
+  } = useBeacons();
+  const {privacy_policy_url} = useRemoteConfig();
+  const style = useStyle();
+  const {clearHistory} = useSearchHistory();
+  const [isCleaningCollectedData, setIsCleaningCollectedData] =
+    React.useState<boolean>(false);
+  const [isCleaningHistoryData, setIsCleaningHistoryData] =
+    React.useState<boolean>(false);
```

Comment:
Does these actions need a loading spinner? Especially the one with cleaning history (which is just local on the device) should be pretty immediate. The general notion is to add a loading spinner if anything takes a significant time for the user. 

A couple articles on the matter:

https://ux.stackexchange.com/questions/92092/what-is-the-minimum-waiting-time-after-when-should-a-page-loader-be-displayed-to
https://uxmovement.com/navigation/progress-bars-vs-spinners-when-to-use-which